### PR TITLE
initial commit of v4

### PR DESCRIPTION
**Pull request by jonathaningram**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,53 +1,34 @@
...
-    "gen:useragent": "node scripts/genuseragent",
-    "docs": "typedoc ./src/index.ts --mode file --out ./docs --exclude \"**/*+(.test).ts\"",
-    "prepublishOnly": "npm-run-all build"
-  },
   "dependencies": {
-    "debug": "^4.1.1",
-    "deepmerge": "^4.0.0",
-    "grpc": "^1.22.2",
-    "protobufjs": "^6.8.8"
+    "request": "2.88.2",
```
Comment (sampotts) [inline package.json:16]: Any concerns with this being deprecated? 
https://www.npmjs.com/package/request
https://github.com/request/request/issues/3142

File: package.json [hunk 2]
```diff
@@ -1,53 +1,34 @@
...
   "name": "@sajari/sdk-node",
-  "version": "2.0.0-beta.7",
-  "main": "dist/index.js",
-  "types": "dist/index.d.ts",
-  "author": "Ben Hinchley <bhinchley@sajari.com>",
+  "version": "4.0.0-alpha.1",
+  "repository": "git@github.com:github.com/sajari/sdk-node.git",
+  "author": "Sajari Pty Ltd",
   "license": "MIT",
+  "main": "./build/src/index.js",
```
Comment (sampotts) [inline package.json:7]: Is it worth us adding ESM support at this stage while we're doing a big change?
https://github.com/standard-things/esm

File: package.json (hunk 1 above)
Comment (jonathaningram) [reply package.json:16]: Unfortunately the Node SDK is generated at this stage and it uses `request`. There's no reason we cannot change it later though—it's hopefully an internal detail that doesn't leak to users.

The other option is that we will just write our own template for the generator. Or modify the existing one.

File: package.json (hunk 2 above)
Comment (jonathaningram) [reply package.json:7]: Yep. Just need to get the MVP out though to begin with!

Comment (sampotts) [reply package.json:7]: Cool, yeah I guess it might be nice to do in `4.1.0`?

Comment (jonathaningram) [reply package.json:7]: Yeah even 4.0 might be possible. I'm only looking to get a v4 alpha or beta tag out so there may be time to do it anyway.

**Pull request merged**